reporter.py

```diff
@@ -58,21 +53,16 @@ def get_char_summary(lot, connection):
 
 
 def get_antibody_approval(antibody, target, connection):
-        search = encodedcc.get_ENCODE('search/?searchTerm='+antibody+'&type=antibody_approval', connection, frame='embedded')
-        for approval in search['@graph']:
-            if approval['target']['name'] == target:
-                return approval['status']
-        return "UNKNOWN"
+    search = encodedcc.get_ENCODE(
+        'search/?searchTerm=' + antibody + '&type=antibody_approval', connection, frame='embedded')
+    for approval in search['@graph']:
+        if approval['target']['name'] == target:
+            return approval['status']
+    return "UNKNOWN"
 
```

Comment:
since you are list comprehensions, I think this could be rewritten as:
(untested code)
statuses = set([ a['status'] for a in search['@graph'] if a['target']['name'] == target ])
if len(statuses) == 1:
    return statuses[0]
else: 
   # ERROR
   return "UNKNOWN" # or somer other error message

Also naked dict keys like approval['target']['name'] are likely to throw KeyErrors unless all these properties are requrired in the schema.   You could use approval.get('target',{}).get('name',"") if you are not sure.

Reply:
> Also naked dict keys like approval['target']['name'] are likely to throw KeyErrors

Didn't write this, didn't want to touch it. The whole file is like this, needs to be rewritten.